tests/unit/s2n_record_test.c

```diff
@@ -429,6 +429,34 @@ int main(int argc, char **argv)
         EXPECT_EQUAL(s2n_stuffer_data_available(&conn->header_in), 3);
     };
 
+    /* Record version is recorded for the first message received (Client Hello) */
+    {
```

Comment:
Do we have a test that makes sure s2n_connection_free_handshake doesn't cause reading records to fail? Touching the ClientHello while potentially receiving ApplicationData is unexpected. We've discussed moving the whole CH to the heap before, not just the raw bytes, and if we did that then checking a field on the CH after the handshake has completed and been freed could cause issues.

But maybe that's a problem for future us.

Reply:
I think we should be good here. [We have a tests that check recv/send right after freeing the handshakes. ](https://github.com/aws/s2n-tls/blob/1b7616726ef9f55fd1b8224402e38a9a8e4624bc/tests/unit/s2n_connection_test.c#L671) So if we move the client hello to the heap this test should catch any issues. But yeah, good callout, it's certainly weird what I am doing here.